### ISSUE TITLE
Issues about storage resources not loading

### ISSUE THREAD

**Issue by marktopper** [thedevdojo/voyager#834]
If you have problems with storage resources not loading, please follow this guide for opening an issue.
All issues about this that does not follow this guide will be closed.

### You have tried
1. Updating Voyager to the latest version.
2. You have already ran `php artisan storage:link`.

## Information we need
1. Laravel version.
2. Voyager version.
3. The Voyager storage configuration
4. The Laravel Filesystems disk configuration
5. The full URL of the page that gives this error (you may replace the domain with xx.xx)
6. The full URL of the resource that the pages fails to load (you may replace the domain with xx.xx)
7. The location on where the requested file is located.

**Comment by troix1** [comment]
My Images in admin panel do not show up, information below:

**Laravel version: 5.3**
**Voyager version: 0.11.3**
**Voyager storage: Everything default**
**Laravel Filesystems disk config: default**
**URL:** xx.xx/admin and all of the BREAD xx.xx/admin/bread
**URL of the resource:** public/storage/users/January2017/Esal0SYtaoZdPZSJ2l8h.jpg and others

*Project is hosted in shared hosting, to get it to work I had to make a change to .htaccess, that fixed the front end, but looks like it ruined the admin panel.*

**.htaccess lines added/changed:**
DirectoryIndex public/index.php
RewriteRule ^ public/index.php [L]

Also, at first Voyager assets weren't working so I changed assets_path and added a **"/public/"** before the 
**'assets_path' => '/public/vendor/tcg/voyager/assets',**

Symlink is created and works as intended i think

**Comment by marktopper** [comment]
troix1: Where is the file `Esal0SYtaoZdPZSJ2l8h.jpg` located at on your server?

**Comment by troix1** [comment]
/public_html/public/storage/users/January2017/

**Comment by marktopper** [comment]
After what I can see, then you tell me that the file is requested in the same folder that it exists in.
Are you running on other ports than 80?

**Comment by troix1** [comment]
It is running on port 80. I have another example. Images in BREAD - browse:
**Image location:** /public_html/ **public/** storage/frontpages/February2017/QcRaTnMUgN3HEa6gwDx9.jpg"
**Requesting URL:** src="/storage/frontpages/February2017/QcRaTnMUgN3HEa6gwDx9.jpg"

It is missing the public/ in url

**Comment by marktopper** [comment]
@troix1. I see your problem now.
The domain root path for Laravel should be set to be `public` folder, so instead you will now have to access the file via the URL at `/public/storage/frontpages/February2017/QcRaTnMUgN3HEa6gwDx9.jpg` instead of `/storage/frontpages/February2017/QcRaTnMUgN3HEa6gwDx9.jpg`.
This will have serious security issues on your site, however if you can not set the `public` folder to be the domain root, you may have to open up `config/filesystems.php` and file the configuration for the public disk and change the visibility path (https://github.com/laravel/laravel/blob/master/config/filesystems.php#L55) to include the starting `public/` part of the URL.

**Comment by troix1** [comment]
Good morning and thanks for your help @marktopper , I have tried to set up my host folders in two ways, the one I mentioned was to copy all the Laravel data into ```public_html```, the other one I have set up that ```public_html``` only contains the Laravel's ```public``` folder and the other files are in folder ```Laravel```, does that make any difference security wise and could I get it to work that way?

On both domains I can't get images to display in ```xx.xx/admin```, I have looked into ```Filesystems.php``` and to be honest I do not know what to do, looks like this:
```
        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'visibility' => 'public',
        ],
```
 I added this line ```'url' => env('APP_URL').'/public/storage',``` and cleared config and cache, nothing changed. Then I changed ```'root' => '/public_html/public',``` but nothing seems to do anything, I could use a bit more help please.

**Comment by luismareze** [comment]
I am getting the same situation in here.
The same settings I am using.

**Comment by marktopper** [comment]
Just to notice users viewing this thread. Than the issue @troix1 had was solved at #815.

**Comment by fletch3555** [comment]
The correct solution here should be to move the project up one directory.  the `public` folder should **BE** the `public_html` folder, not be within it.  The app folder should be adjacent to the public_html folder.  That would fix most of this.  It's a sysadmin issue, not a Voyager issue.
